terror_encoding.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+)
+
+type errorRender struct {
+	// Deprecated field, please use `RFCCode` instead.
+	Class       int    `json:"class" toml:"-"`
+	Code        int    `json:"code" toml:"-"`
+	Msg         string `json:"message" toml:"message"`
+	RFCCode     string `json:"rfccode" toml:"code"`
+	Description string `json:"description,omitempty" toml:"description"`
+	Workaround  string `json:"workaround,omitempty" toml:"workaround"`
+}
+
+// RenderTOML implements MarshaText and UnmarshaText for Error.
+// Outputs text in TOML format.
+type RenderTOML Error
+
+var _ encoding.TextMarshaler = RenderTOML{}
+var _ encoding.TextUnmarshaler = &RenderTOML{}
+
+// MarshalText implements encoding.TextMarshaler interface.
+func (rd RenderTOML) MarshalText() ([]byte, error) {
+	e := Error(rd)
+	render := &errorRender{
+		Msg:         e.GetMsg(),
+		Description: e.description,
+		Workaround:  e.workaround,
+		RFCCode:     string(e.codeText),
+	}
+	buf := new(bytes.Buffer)
+	if err := toml.NewEncoder(buf).Encode(render); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler interface.
+func (rd *RenderTOML) UnmarshalText(data []byte) error {
```

Comment:
Why not `*Error` directly, but use `RenderTOML` and `RenderJSON`

Reply:
The UnmarshalText has been removed.